src/System.Net.Sockets/tests/FunctionalTests/SocketOptionNameTest.cs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Net.Test.Common;
+using System.Runtime.InteropServices;
+
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public class SocketOptionNameTest
+    {
+        private static bool IsWindows10Platform
+        {
+            get
+            {
+                return Capability.IsWindows10Platform();
+            }
+        }
+
+        private static bool IsNotWindows10Platform
+        {
+            get
+            {
+                return !Capability.IsWindows10Platform();
+            }
+        }
+
+        [ConditionalFact("IsNotWindows10Platform")]
+        public void ReuseUnicastPort_CreateSocketGetOption_NotWindows10Platform_Throws()
+        {
+            var socket = new Socket(AddressFamily.InterNetwork, SocketType.Stream, ProtocolType.Tcp);
+            
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
> On non-Windows10, it throws SocketException when get or set the socket option.
>
> On *NIX systems, it throws PlatformNotSupportedException.

Hm, we might want to change the *nix behavior to at least throw a `SocketException`. Please file a bug on me to look into this.